crates/goose/src/agents/truncate.rs

```diff
@@ -226,7 +226,10 @@ impl Agent for TruncateAgent {
                     &messages,
                     &tools,
                 ).await {
-                    Ok((response, usage)) => {
+                    Ok((mut response, usage)) => {
+                        // Post-process / structure the response
+                        response = capabilities.provider().structure_response(response, &tools).await?;
```

Comment:
i think this line would break if some uses anthropic?

Reply:
i think we only need to run this if `GOOSE_TOOLSHIM=true | 1` - can add that check here